### DISCUSSION TITLE
trying to get the heap object to be zero but once gnet is ran, it will give 2 heap object per second

### DISCUSSION THREAD

**Discussion by sprappcom** [panjf2000/gnet#597] [Q&A] [answered]
the 2 heap object / second is generated by runeventloops? if idle, it will do nothing but once ran, after my 
wrk tested on http, it will generate 2 heap object per second.

how to make it stop generating 2 heap object every second when idle?

```
go tool pprof -top heap.prof
File: httpx7
Type: inuse_space
Time: Apr 26, 2024 at 3:17am (EDT)
Showing nodes accounting for 2178.67kB, 100% of 2178.67kB total
      flat  flat%   sum%        cum   cum%
 2178.67kB   100%   100%  2178.67kB   100%  github.com/panjf2000/gnet/v2.(*engine).runEventLoops
         0     0%   100%   544.67kB 25.00%  github.com/leslie-fei/gnettls.Run (inline)
         0     0%   100%  2178.67kB   100%  github.com/panjf2000/gnet/v2.(*engine).start
         0     0%   100%  2178.67kB   100%  github.com/panjf2000/gnet/v2.Run
         0     0%   100%  2178.67kB   100%  github.com/panjf2000/gnet/v2.run
         0     0%   100%     1634kB 75.00%  main.main.func1
         0     0%   100%   544.67kB 25.00%  main.main.func2
```

**Comment by panjf2000** [comment]
I can't really see the "2 heap objects" in your pprof data.

Also, could you do the pprof without TLS, I'd like to see the result of that.

**Reply by sprappcom** [reply, marked as answer]
@panjf2000 so sorry my bad, i'm truly sorry for this message. i just figured out where the issue is with my code. gnet is great. no such issue. thx. havent tested the tls. will do so.

**Comment by panjf2000** [comment]
Since everything is fine, let's close this.
